testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -617,4 +617,32 @@ public void testSetViewAuthorizationWithSecurityInvoker()
     {
         assertQuerySucceeds("ALTER VIEW mock.default.test_view_invoker SET AUTHORIZATION some_other_user");
     }
+
+    @Test
+    public void testUseStatementAccessControl()
+    {
+        executeExclusively(() -> {
+            Session session = testSessionBuilder()
+                    .setCatalog(Optional.empty())
+                    .setSchema(Optional.empty())
+                    .build();
+            getQueryRunner().execute(session, "USE tpch.tiny");
+
+            assertThatThrownBy(() -> getQueryRunner().execute("USE not_exists_catalog.tiny"))
+                    .hasMessageMatching("Catalog does not exist: not_exists_catalog");
+            assertThatThrownBy(() -> getQueryRunner().execute("USE tpch.not_exists_schema"))
+                    .hasMessageMatching("Schema does not exist: tpch.not_exists_schema");
+
+            try {
+                getQueryRunner().getAccessControl().denyCatalogs(catalog -> !catalog.equals("tpch"));
+                assertThatThrownBy(() -> getQueryRunner().execute("USE tpch.tiny"))
+                        .hasMessageMatching("Access Denied: Cannot access catalog tpch");
+                assertThatThrownBy(() -> getQueryRunner().execute("USE tpch.not_exists_schema"))
+                        .hasMessageMatching("Access Denied: Cannot access catalog tpch");
```

Comment:
There also needs to be a test that allows access to catalog but denies access to a schema

Reply:
Is it possible to add such a test?

Reply:
Added